packages/client-sdk-nodejs/src/config/retry/storage-default-eligibility-strategy.ts

```diff
@@ -0,0 +1,104 @@
+import {MomentoLogger, MomentoLoggerFactory} from '../../';
+import {Status} from '@grpc/grpc-js/build/src/constants';
+import {
+  EligibilityStrategy,
+  EligibleForRetryProps,
+} from './eligibility-strategy';
+
+const retryableGrpcStatusCodes: Array<Status> = [
+  // including all the status codes for reference, but
+  // commenting out the ones we don't want to retry on for now.
+
+  // Status.OK,
+  // Status.CANCELLED,
+  // Status.UNKNOWN,
+  // Status.INVALID_ARGUMENT,
+  Status.DEADLINE_EXCEEDED,
+  // Status.NOT_FOUND,
+  // Status.ALREADY_EXISTS,
+  // Status.PERMISSION_DENIED,
+  // Status.RESOURCE_EXHAUSTED,
+  // Status.FAILED_PRECONDITION,
+  // Status.ABORTED,
+  // Status.OUT_OF_RANGE,
+  // Status.UNIMPLEMENTED,
+  Status.INTERNAL,
+  Status.UNAVAILABLE,
+  // Status.DATA_LOSS,
+  // Status.UNAUTHENTICATED
+];
+
+const retryableRequestTypes: Array<string> = [
+  '/store.Store/Put',
+  '/store.Store/Get',
+  '/store.Store/Delete',
+];
+
+export class DefaultStorageEligibilityStrategy implements EligibilityStrategy {
+  private readonly logger: MomentoLogger;
+
+  constructor(loggerFactory: MomentoLoggerFactory) {
+    this.logger = loggerFactory.getLogger(this);
+  }
+
+  isEligibleForRetry(props: EligibleForRetryProps): boolean {
+    if (!retryableGrpcStatusCodes.includes(props.grpcStatus.code)) {
+      this.logger.debug(
+        `Response with status code ${props.grpcStatus.code} is not retryable.`
+      );
+      return false;
+    }
+
+    const errorMetadata = props.grpcStatus.metadata.get('err');
+    // Expecting only one error condition in metadata
+    if (errorMetadata.length === 1) {
+      switch (errorMetadata[0].toString()) {
+        case 'momento_general_err':
+          return false;
+        case 'server_is_busy': {
+          // Retry disposition will show up only for err metadata "server_is_busy"
+          const retryMetadata =
+            props.grpcStatus.metadata.get('retry_disposition');
+          if (retryMetadata.length === 1) {
+            const retryDisposition = retryMetadata[0].toString();
+            switch (retryMetadata[0].toString()) {
+              case 'retryable':
+                // Retryable = could retry soon (default 100ms)
+                return true;
+              case 'possibly_retryable':
+                // Possibly retryable = could retry in a second or two
+                return true; // return a value here instead?
```

Comment:
asked Ben what sdks should default to doing when "possibly_retryable" is received and he said it means the server is likely in maintenance, could retry the request in 1-2 seconds, but currently we return only true/false from this method and default to retrying in 100ms if true

Reply:
this doesn't match the docstrings in his client-protos PR, nor my expectations about these values based on our previous discussions:

https://github.com/momentohq/client-protos/blob/2038613adca37ef57cfd81dd7c1539a2d570bbc3/proto/store.proto#L38

my belief is that the server would always be sending `retryable` when in maintenance mode, and if we get `possibly_retryable` it means it's not safe to retry unless the operation is idempotent or the user has configured us to retry on idempotent operations.

Reply:
revised the metadata parsing and removed the "err" metadata checking (it appears irrelevant to determining retries since the error code checking occurs first)